sdk/identity/Azure.Identity/README.md

```diff
@@ -1,99 +1,175 @@
+var credential = new CertificateCredential(tenantId, clientId, certificate);
 
-```Python
-thing = client.get_thing(id)
+var secretClient = new SecretClient(new Uri("https://myvault.azure.vaults.net/"), credential);
 ```
 
-### List the things
+## Chaining Credentials
+
+The `ChainedTokenCredential` class provides the ability to link together multiple credential instances to be tried sequential when authenticating. The following example demonstrates creating a credential which will attempt to authenticate using managed identity, and fall back to certificate authentication if a managed identity is unavailable in the current environment.
+```c#
+using Azure.Identity;
+using Azure.Messaging.EventHubs;
+
+var managedCredential = new ManagedIdentityCredential(clientId);
 
-Use [list_things](not-valid-link) to get one or more Thing objects from the service. If there are no Things available, a `404` exception is thrown (see [Troubleshooting](#troubleshooting) for details on handling exceptions).
+var certCredential = new CertificateCredential(tenantId, clientId, certificate);
 
-```Python
-things = client.list_things()
+// authenticate using managed identity if it is available otherwise use certificate auth
+var credential = new ChainedTokenCredential(managedCredential, certCredential);
+
+var eventHubClient = new EventHubClient("myeventhub.eventhubs.windows.net", "myhubpath", credential);
 ```
 
 ## Troubleshooting
 
-Describe common errors and exceptions, how to "unpack" them if necessary, and include guidance for graceful handling and recovery.
-
-Provide information to help developers avoid throttling or other service-enforced errors they might encounter. For example, provide guidance and examples for using retry or connection policies in the API.
+Errors arising from authentication can be raised on any service client method which makes a request to the service. This is because the first time the token is requested from the credential is on the first call to the service, and any subsequent calls might need to refresh the token. In order to distinguish these failures from failures in the service client Azure Identity classes raise the `AuthenticationFailedException` with details to the source of the error in the exception message as well as possibly the error message.
 
-If the package or a related package supports it, include tips for logging or enabling instrumentation to help them debug their code.
+For more details on dealing with errors arising from failed requests to Azure Active Directory, or managed identity endpoints please refer to the Azure Active Directory [documentation on authorization error codes][aad_err_doc].
 
 ## Next steps
+Currently the following client libraries support authenticating with `TokenCredential` and the Azure Identity library.  Additional documentation on use of these client libraries along samples with can be found in the links below.
```

Comment:
This text sounds like it belongs more to `Additional Documentation` instead of `Next Steps`